cmd/booster-bitswap/filters/configfilter.go

```diff
@@ -0,0 +1,174 @@
+package filters
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/dustin/go-humanize"
+	"github.com/filecoin-project/boost/cmd/booster-bitswap/requestcounter"
+	"github.com/ipfs/go-cid"
+	peer "github.com/libp2p/go-libp2p/core/peer"
+)
+
+// PeerListType is either an allow list or a deny list
+type PeerListType string
+
+// AllowList is a peer list where only the specified peers are allowed to serve retrievals
+const AllowList PeerListType = "allowlist"
+
+// DenyList is a peer list where the specified peers cannot serve retrievals, but all others can
+const DenyList PeerListType = "denylist"
+
+// BandwidthMeasure provides an up to date measurement of the current bytes per second transferred
+type BandwidthMeasure interface {
+	AvgBytesPerSecond() uint64
+}
+
+type RequestCounter interface {
+	StateForPeer(p peer.ID) requestcounter.ServerState
+}
+
+type remoteConfig struct {
+	peerListType                   PeerListType
+	peerList                       map[peer.ID]struct{}
+	underMaintenance               bool
+	maxSimultaneousRequests        uint64
+	maxSimultaneousRequestsPerPeer uint64
+	maxBandwidth                   uint64
+}
+
+// ConfigFilter manages filtering based on a remotely fetched retrieval configuration
+type ConfigFilter struct {
+	remoteConfigLk   sync.RWMutex
+	bandwidthMeasure BandwidthMeasure
+	requestCounter   RequestCounter
+	remoteConfig     remoteConfig
+}
+
+// NewConfigFilter constructs a new peer filter
+func NewConfigFilter(bandwidthMeasure BandwidthMeasure, requestCounter RequestCounter) *ConfigFilter {
+	return &ConfigFilter{
+		bandwidthMeasure: bandwidthMeasure,
+		requestCounter:   requestCounter,
+		remoteConfig: remoteConfig{
+			peerListType:                   DenyList,
+			peerList:                       make(map[peer.ID]struct{}),
+			underMaintenance:               false,
+			maxSimultaneousRequests:        0,
+			maxSimultaneousRequestsPerPeer: 0,
+			maxBandwidth:                   0,
+		},
+	}
+}
+
+// FulfillRequest checks if a given peer is in the allow/deny list and decides
+// whether to fulfill the request
+func (cf *ConfigFilter) FulfillRequest(p peer.ID, c cid.Cid) (bool, error) {
```

Comment:
I think we should add some instrumentation around this function in terms of logs and metrics (fulfilled and blocked counters).

I am pretty sure someone will mess up the configuration and then it will be very hard to figure out if/why their node is not serving requests. We should be able to turn on `debug` logging and see why requests are blocked/passed through.